src/components/dashjs-api-control-element/dashjs-api-control-element.tsx

```diff
@@ -65,57 +46,85 @@ export class DashjsAPIControlElement {
       });
       toast.present();
     } else {
-      this.valueChanged.emit(this.functionValue);
+      this.valueChanged.emit(functionValue);
     }
   }
 
+  setAndSave(index, value) {
+    functionValue[index] = value;
+    LocalStorage.updateKeyInKeyValueObject(STRING_API_ELEM, this.name, JSON.stringify(functionValue));
```

Comment:
just from an architectural component perspective wouldn't it be better to have all the localStorage handling in one component? (via the emitted changed value) Because this way you may set yourself up for trouble... deleting in one component and writing in another seems like a bad idea.